eventsourcing/src/main/java/org/axonframework/eventsourcing/GenericAggregateFactory.java

```diff
@@ -50,14 +54,30 @@
      *                                        constructor.
      */
     public GenericAggregateFactory(Class<T> aggregateType) {
-        super(aggregateType);
+        this(AnnotatedAggregateMetaModelFactory.inspectAggregate(aggregateType));
         Assert.isFalse(Modifier.isAbstract(aggregateType.getModifiers()), () -> "Given aggregateType may not be abstract");
-        try {
-            this.constructor = ensureAccessible(aggregateType.getDeclaredConstructor());
-        } catch (NoSuchMethodException e) {
-            throw new IncompatibleAggregateException(format("The aggregate [%s] doesn't provide a no-arg constructor.",
-                                                            aggregateType.getSimpleName()), e);
-        }
+    }
+
+    /**
+     * Initialize the AggregateFactory for creating instances of the given {@code aggregateModel}.
+     *
+     * @param aggregateModel the model of aggregate this factory creates instances of
+     */
+    public GenericAggregateFactory(AggregateModel<T> aggregateModel) {
+        super(aggregateModel);
+        aggregateModel.types()
+                      .filter(type -> !Modifier.isInterface(type.getModifiers()))
+                      .filter(type -> !Modifier.isAbstract(type.getModifiers()))
+                      .forEach(type -> {
+            try {
+                Constructor<?> constructor = ensureAccessible(type.getDeclaredConstructor());
+                constructors.put(type, constructor);
+            } catch (NoSuchMethodException e) {
+                throw new IncompatibleAggregateException(format(
+                        "The aggregate (or some of its subtypes) [%s] doesn't provide a no-arg constructor.",
```

Comment:
Can't we mention the actual type where the No-Arg Constructor couldn't be found? That would help the developer.